pkg/app/cmd.go

```diff
@@ -123,20 +128,6 @@ func runE(
 		}
 	}
 
-	// instantiate the tester if testing was specified
-	// NOTE: we need to do this before building etc, so we can fail fast on
-	// invalid options to the tester
-	var tester types.Tester
-	if newTester != nil {
-		t, err := newTester(opts, testerArgs, deployer)
```

Comment:
Removing this removes an opportunity for a runtime check of tester options before it is actually executed. This could put the user in a situation where the deployer goes through build and up and _then_ the tester errors because of a bad flag/option. I'm specifically thinking of the artifacts dir change in #8, but this could impact other future flags as well.

Reply:
Not saying it necessarily has to be checked before the deployer is called, but it would help the UX.

Reply:
Yes, I think that is an expected side effect. I'm planning to make `--help` a requirement for all testers, in which case we could overload it's functionality to also ensure all flags are parsed.

Reply:
Okay, I added a fix for this behavior since this seems pretty important UX.